src/openapi/streaming/connection/transport/signalr-core-transport.ts

```diff
@@ -370,13 +371,14 @@ class SignalrCoreTransport implements StreamingTransportInterface {
 
         // close message stream before closing connection
         if (this.messageStream) {
-            return (
-                this.messageStream
+            const closePromise =
+                ( this.connection && this.connection.state === HubConnectionState.Connected)
                     // @ts-expect-error cancelCallback is no included in the IStreamResult but according to implementation it exists
-                    .cancelCallback()
-                    .then(sendCloseMessage)
-                    .then(() => this.connection && this.connection.stop())
-            );
+                    ? this.messageStream.cancelCallback()
+                    : Promise.resolve();
+            return closePromise
+                .then(sendCloseMessage)

```

Comment:
won't send fail if we are not currently connected?
I wonder if we guard this too with checking the connection state?

Reply:
good catch, connection.invoke calls sendMessage as well